lib/commands/new/command.js

```diff
@@ -1,47 +1,226 @@
     );
+  }
+
+  async confirmToWriteOnExistingFolder(projectName) {
+    let question;
+    let yesHint;
+
+    if (!this.here && fs.isFile(projectName)) {
+      throw new Error(`There is an existing file named as ${projectName}, cannot create folder ${projectName}.`);
+    } else if (!this.here && fs.isDirectory(projectName)) {
+      if (this.unattended) {
+        throw new Error(`The '${projectName}' folder already exists. Please make sure it's not there for unattended mode.`);
+      }
+      question = `WARNING: The '${projectName}' folder already exists. Would you like to create the project in this folder?`;
+      yesHint = `Continue even though the '${projectName}' directory already exists.`;
+    } else if (this.here && fs.readdirSync(process.cwd()).length > 0) {
+      // Default answer is yes in unattended mode
+      if (this.unattended) return true;
+      question = 'WARNING: The current directory is not empty. Would you like to create the project in this folder?';
+      yesHint = 'Continue even though the current directory is not empty.';
+    }
+
+    if (!question) return true;
+
+    const confirm = await this.ui.question(
+      c.red(question),
+      [
+        {value: 'abort', message: 'Abort', hint: 'Aborts the new project wizard.'},
+        {value: 'continue', message: 'Continue', hint: c.red(yesHint)}
+      ]
+    );
+
+    return confirm === 'continue';
+  }
 
-    let state = {};
+  async showSummary(projectName, features) {
+    // Ensure to wrap projectName with quotes if there is white space in it.
+    const wrappedName = projectName.match(/\s/) ? JSON.stringify(projectName) : projectName;
+    const feats = await minFeatures(features, this.plugin);
 
-    if (this.options.hasFlag('here')) {
-      state.name = this.options.originalBaseDir;
-      state.defaultOrCustomOverride = 'custom';
+    await this.ui.log('');
+    await this.logTitle('📖', 'Project Summary');
+    await this.logBody(`This project was bootstrapped based on following selected features:
+${c.green(features.join(' '))}
+Next time, you can generate a new project faster in unattended mode with the same features:
+${cmd(`au new ${wrappedName} --unattended${feats.length ? ' --select ' : ''}${feats.join(',')}${this.plugin ? ' --plugin' : ''}${this.here ? ' --here' : ''}`)}
+Or use the shorter form:
+${cmd(`au new ${wrappedName} -u${feats.length ? ' -s ' : ''}${feats.join(',')}${this.plugin ? ' -p' : ''}${this.here ? ' -h' : ''}`)}
+Note the command might not supply all the selected features, as some of them could be from default offering.`);
+  }
 
-      if (state.name.indexOf('/') !== -1) {
-        let parts = state.name.split('/');
-        state.name = parts[parts.length - 1];
-      } else if (state.name.indexOf('\\') !== -1) {
-        let parts = state.name.split('\\');
-        state.name = parts[parts.length - 1];
+  async optionallyInstallDeps(packageManager, projectFolder) {
+    await this.logTitle('🚚', 'Project Dependencies');
```

Comment:
wow what's this :)

Reply:
is this going to work in every terminal?

Reply:
Nope, emoji (utf8) does not print out properly on Windows PowerShell, but it did not hurt much. I am open to remove them, but they are cool :-)

Reply:
I like them too, but not if they do not work everywhere

Reply:
Removed.